geopandas/array.py

```diff
@@ -362,62 +435,119 @@ def shape(self):
         -------
         shape : tuple
         """
-
         return tuple([len(self)])
 
-    def ravel(self, order='C'):
-        """ Return a flattened (numpy) array.
+    def to_dense(self):
+        """Return my 'dense' representation
 
         For internal compatibility with numpy arrays.
 
         Returns
         -------
-        raveled : numpy array
+        dense : array
         """
-        return np.array(self)
+        return to_shapely(self.data)
 
-    def view(self):
-        """Return a view of myself.
+    def isna(self):
+        """
+        Boolean NumPy array indicating if each value is missing
+        """
+        return self.data == 0
 
-        For internal compatibility with numpy arrays.
+    def unique(self):
+        """Compute the ExtensionArray of unique values.
 
         Returns
         -------
-        view : Categorical
-           Returns `self`!
+        uniques : ExtensionArray
         """
-        return self
+        from pandas import factorize
+        _, uniques = factorize(self)
+        return uniques
 
-    def to_dense(self):
-        """Return my 'dense' representation
+    @property
+    def nbytes(self):
+        return self.data.nbytes
 
-        For internal compatibility with numpy arrays.
+    # ExtensionArray specific
+
+    @classmethod
+    def _constructor_from_sequence(cls, scalars):
+        """Construct a new ExtensionArray from a sequence of scalars.
 
+        Parameters
+        ----------
+        scalars : Sequence
+            Each element will be an instance of the scalar type for this
+            array, ``cls.dtype.type``.
         Returns
         -------
-        dense : array
+        ExtensionArray
         """
-        return to_shapely(self.data)
+        return from_shapely(scalars)
 
-    def get_values(self):
-        """ Return the values.
+    @classmethod
+    def _from_factorized(cls, values, original):
+        """Reconstruct an ExtensionArray after factorization.
 
-        For internal compatibility with pandas formatting.
+        Parameters
+        ----------
+        values : ndarray
+            An integer ndarray with the factorized values.
+        original : ExtensionArray
+            The original ExtensionArray that factorize was called on.
+
+        See Also
+        --------
+        pandas.factorize
+        ExtensionArray.factorize
+        """
+        return from_wkb(values)
```

Comment:
This implementation seems surprising given the method title.

Reply:
Ah I see, we represent geometries with the appropriate bytestring

Reply:
Yes, for now I went for using WKB as an intermediate representation for unique / factorize (and possibly also for sorting). With the idea that WKB is the cheapest representation to generate that can be used for algorithms like unique and factorize, without implementing them ourselves (in principle we could also code them in cython using the geos interface for equality and populating a table)